client/src/components/shared/SaveBar.tsx

```diff
@@ -0,0 +1,76 @@
+// SPDX-FileCopyrightText: 2023 The Aalto Grades Developers
+//
+// SPDX-License-Identifier: MIT
+
+import {ClearRounded, SaveOutlined} from '@mui/icons-material';
+import {Box, Button, Fade, Typography, useTheme} from '@mui/material';
+const SaveBar = ({
+  show,
+  handleSave,
+  handleDiscard,
+  loading,
+  disabled,
+}: {
+  show: boolean;
+  loading?: boolean;
+  disabled?: boolean;
+  handleSave?: Function;
+  handleDiscard: Function;
+}): JSX.Element => {
```

Comment:
`@deprecated — Use React.JSX instead of the global JSX namespace.`.
I recommend just doing    `import {JSX} from 'react'` instead of using `React.JSX`.